src/generators/legacy-json/utils/buildSection.mjs

```diff
@@ -0,0 +1,385 @@
+    .map(node =>
+      entry.rawContent.slice(
+        node.position.start.offset,
+        node.position.end.offset
+      )
+    )
+    .join('')
+    .replace(/\s+/g, ' ')
+    .replaceAll(/<!--.*?-->/gs, '');
+
+  if (!current.textRaw) {
+    throw new Error(`Empty list item: ${JSON.stringify(child)}`);
+  }
+
+  let text = current.textRaw;
+
+  // Extract name
+  if (RETURN_EXPRESSION.test(text)) {
+    current.name = 'return';
+    text = text.replace(RETURN_EXPRESSION, '');
+  } else {
+    const [, name] = text.match(NAME_EXPRESSION) || [];
+    if (name) {
+      current.name = name;
+      text = text.replace(NAME_EXPRESSION, '');
+    }
+  }
+
+  // Extract type (if provided)
+  const [, type] = text.match(TYPE_EXPRESSION) || [];
+  if (type) {
+    current.type = type;
+    text = text.replace(TYPE_EXPRESSION, '');
+  }
+
+  // Remove leading hyphens
+  text = text.replace(LEADING_HYPHEN, '');
+
+  // Extract default value (if exists)
+  const [, defaultValue] = text.match(DEFAULT_EXPRESSION) || [];
+  if (defaultValue) {
+    current.default = defaultValue.replace(/\.$/, '');
+    text = text.replace(DEFAULT_EXPRESSION, '');
+  }
+
+  // Add remaining text to the desc
+  if (text) {
+    current.desc = text;
+  }
+
+  const options = child.children.find(child => child.type === 'list');
+  if (options) {
+    current.options = options.children.map(child =>
+      parseListItem(child, entry)
+    );
+  }
+
+  return current;
+}
+
+/**
+ * @param {import('../types.d.ts').HierarchizedEntry} entry
+ * @param {import('../types.d.ts').Section} parentSection
+ */
+function handleEntry(entry, parentSection) {
+  // Clone the children so we don't mess with any other generators
+  let [headingNode, ...nodes] = structuredClone(entry.content.children);
+
+  /**
+   * Grabs stability number & text and adds it to the section
+   * @param {import('../types.d.ts').Section} section
+   */
+  const parseStability = section => {
+    let needsCompressing = false;
+
+    // Remove metadata not directly the inferable from the markdown
+    nodes.forEach((node, i) => {
+      if (
+        node.type === 'blockquote' &&
+        node.children.length === 1 &&
+        node.children[0].type === 'paragraph' &&
+        nodes.slice(0, i).every(node => node.type === 'list')
+      ) {
+        const text = transformNodesToString(node.children[0].children);
+        const stability = /^Stability: ([0-5])(?:\s*-\s*)?(.*)$/s.exec(text);
+        if (stability) {
+          section.stability = parseInt(stability[1], 10);
+          section.stabilityText = stability[2].replaceAll('\n', ' ').trim();
+
+          delete nodes[i];
+          needsCompressing = true;
+        }
+      }
+    });
+
+    if (needsCompressing) {
+      // Compress to remove the holes left by deletions
+      nodes = nodes.filter(() => true);
+    }
+  };
+
+  /**
+   *
+   * @param {import('../types.d.ts').Section} section
+   */
+  const parseList = section => {
```

Comment:
This should be a Factory (even a different file) that allows you to handle the signature types for other types.

So that we have tiny methods (that implement an interface) instead of non-ending signatures

Reply:
What do you mean? I'm afraid it's not clicking for me, sorry.